lib/portlayer/storage/image_cache.go

```diff
@@ -232,6 +232,18 @@ func (c *NameLookupCache) WriteImage(op trace.Operation, parent *Image, ID strin
 	return i, nil
 }
 
+func (c *NameLookupCache) Read(op trace.Operation, store *url.URL, child, ancestor string, pathspec map[string]string, data bool) (io.ReadCloser, error) {
+	_, err := util.ImageStoreName(store)
+	if err != nil {
+		return nil, err
+	}
+
+	if archiver, ok := c.DataStore.(Archiver); ok {
+		return archiver.Read(op, store, child, ancestor, pathspec, data)
```

Comment:
Cleaning this up now, to look like:
```go
func (c *NameLookupCache) ReadArchive(op trace.Operation, store *url.URL, child, ancestor string, pathspec map[string]string, data bool) (io.ReadCloser, error) {
        return c.DataStore.ReadArchive(op, store, child, ancestor, pathspec, data)
}
```
The way the storage stuff was implemented, two layers (`NameLookupCache` and `ImageStore`) both implement the `ImageStorer` interface. Both layers appear to use `store` to either check that the particular store (which could be one of many) exists, or to get the store name and make other calls using that. In this case, it is used on the lower layer (`ImageStore`) to get the store name, and then the store name is used to get the datastore path to the vmdks of the appropriate image layers.